pkg/node/statestore.go

```diff
@@ -11,19 +11,46 @@ import (
 
 	"github.com/ethersphere/bee/pkg/logging"
 	"github.com/ethersphere/bee/pkg/statestore/leveldb"
-	"github.com/ethersphere/bee/pkg/statestore/mock"
 	"github.com/ethersphere/bee/pkg/storage"
 	"github.com/ethersphere/bee/pkg/swarm"
+
+	ldb "github.com/syndtr/goleveldb/leveldb"
+	ldbs "github.com/syndtr/goleveldb/leveldb/storage"
 )
 
+type MockStore struct {
+	ldb *ldb.DB
+}
+
+func (m *MockStore) Get(key string, i interface{}) (err error) {
+	return nil
+}
+func (m *MockStore) Put(key string, i interface{}) (err error) {
+	return nil
+}
+func (m *MockStore) Delete(key string) (err error) {
+	return nil
+}
+func (m *MockStore) Iterate(prefix string, iterFunc storage.StateIterFunc) (err error) {
+	return nil
+}
+
+// DB returns the underlying DB storage.
+func (m *MockStore) DB() *ldb.DB {
+	return m.ldb
+}
+func (m *MockStore) Close() error {
+	return nil
+}
+
 // InitStateStore will initialize the stateStore with the given path to the
 // data directory. When given an empty directory path, the function will instead
 // initialize an in-memory state store that will not be persisted.
 func InitStateStore(log logging.Logger, dataDir string) (ret storage.StateStorer, err error) {
 	if dataDir == "" {
-		ret = mock.NewStateStore()
+		ldb, err := ldb.Open(ldbs.NewMemStorage(), nil)
 		log.Warning("using in-mem state store, no node state will be persisted")
-		return ret, nil
+		return &MockStore{ldb: ldb}, err
```

Comment:
I dont think we need to create a separate MockStore... we can use the same leveldb.StateStore, only the underlying levelDB needs to be initialized as in-mem.